ddtrace/contrib/asgi/middleware.py

```diff
@@ -0,0 +1,118 @@
+import sys
+
+from ddtrace import tracer as global_tracer
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.ext import SpanTypes, http
+from ddtrace.http import store_request_headers, store_response_headers
+from ddtrace.propagation.http import HTTPPropagator
+from ddtrace.settings import config
+
+from ...compat import reraise
+from ...internal.logger import get_logger
+from .utils import guarantee_single_callable
+
+log = get_logger(__name__)
+
+config._add("asgi", dict(service_name=config._get_service(default="asgi"), distributed_tracing=True,))
```

Comment:
(Nit) Formatting leftover

```suggestion
config._add("asgi", dict(service_name=config._get_service(default="asgi"), distributed_tracing=True))
```